src/town_cmd.cpp

```diff
@@ -2118,6 +2118,11 @@ static CommandCost TownCanBePlacedHere(TileIndex tile)
 		return_cmd_error(STR_ERROR_TOO_CLOSE_TO_ANOTHER_TOWN);
 	}
 
+	/* Check max height level. */
+	if (GetTileZ(tile) > _settings_game.economy.max_town_heightlevel) {
+		return_cmd_error(STR_ERROR_SITE_UNSUITABLE);
+	}
```

Comment:
Would it be possible to limit this change to only applying during world generation, so that players still retain as much freedom as possible as for where new towns are constructed?

Reply:
Possible yes. I'll leave it up to JGR to decide what the setting for that should look like. Whether it should be somehow in the existing one or an additional flag or something or whether to always allow it. I don't mind either way.

Reply:
I'm inclined to leave it as it. The setting can be easily be changed after map generation by players or the server operator to re-allow town founding at higher levels if that is desired.